### DISCUSSION TITLE
WAV file output max length

### DISCUSSION THREAD

**Discussion by steenl** [pschatzmann/arduino-audio-tools#1053] [Q&A] [answered]
When streaming to WAVEncoder I am finding a maximum audio capture of 4 seconds using both copier.copyN() and the timeout counter method.  Shorter captures to file are done by reducing counter and copyN() value.  
The .wav header content looks valid and increasing info.data_length from default 0x7ff0000 defined in CodecWAV.h, the change is seen in new audio captures, but no effect on the 4 second streaming maximum. 
File size is usually 770K regardless of the attempted sample time.

Is there a maximum audio length supported by WAVencoder, or any workarounds that I can try to extend the audio length captured?

**Comment by pschatzmann** [comment, marked as answer]
WAV files are limited to 4GB in size as they have a 32bit file size header. If you don't specify a length, the header is filled with this 4GB limit and the effective play length is just determined by the effective file size.

What makes you think that there is a 4 seconds limit ? 

You just need to have enough space on the target medium and make sure that the data is saved properly...

**Reply by steenl** [reply]
Agreed, thanks.  My SPIFFS was limiting the output size and verified by expanded SPIFFS size.
Thanks
